### PR TITLE
feat: New group notification

### LINKED ISSUE #3383: feat: notify user when added to a new group
**Is your feature request related to a problem? Please describe.**
Current notification events are -
- star notification
- fork notification
- new assignment notification(WIP)
This will be an addition to main platform improvement of adding more notification events

**Describe the solution you'd like**
- The service logic would have to be improvised to incorporate more notification events in future
-  refine the redirect logic in the controller  
**Describe alternatives you've considered**

**Additional context**
I would like to work on this issue.
**Are you working on this? (Yes/No)**
Yes

### PR DESCRIPTION
**Pull request by VaibhavUpreti**
Fixes #3383 

#### Describe the changes you have made in this PR -
- added feature
- written specs
- Need to work considering PR #3312 to improvise logic in noticed_notification_controller
### Screenshots of the changes (If any) -
(CURRENT VIEW)

Screenshot 2022-12-21 at 12 05 25 AM

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

**Comment by VaibhavUpreti** [comment]
@vedant-jain03 I have to add group name instead of group_id in notification ... and resolve a style issue.
Will ping you after that

**Comment by VaibhavUpreti** [comment]
@vedant-jain03 this one seems to be ready for review....
Final view - 
"user" added to group: "group_name"

Screenshot 2022-12-14 at 8 01 48 PM

File: app/notifications/new_group_notification.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+  def message
+    @group = params[:group][:group_id]
+    group = Group.find(@group)
+    user = group.primary_mentor.name
+    t("users.notifications.new_group_notification", group: group.name, user: user)
+  end
+
+  def icon
+    "far fa-clipboard"
```
[the remark this answers is not in the document]
Comment (vedant-jain03) [reply app/notifications/new_group_notification.rb:14]: Try this icon out!
The style padding is broken for your case as well. Initially, we have used `padding` for different icons. We need a way for that.

For this one, you can use `padding: 12px 10px;`.

File: app/services/notify_user.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
```
Comment (vedant-jain03) [inline app/services/notify_user.rb:2]: You have added this logic in https://github.com/CircuitVerse/CircuitVerse/pull/3312 PR as well. Let @tachyons add his review to that and we can use that as official logic for handling different `notifications` conditionally.

**Comment by VaibhavUpreti** [comment]
@vedant-jain03 
After code review- 
final

File: app/assets/stylesheets/users.scss [hunk 3]
```diff
@@ -501,6 +501,14 @@
     padding: 12px 10px;
   }
 
+  .fa-users {
+    padding: 12px 10px;
+  }
+
+  .fa-clipboard {
+    padding: 11px 10px;
```
Comment (vedant-jain03) [inline app/assets/stylesheets/users.scss:509]:
```suggestion
    padding: 12px 15px;
```

**Review by vedant-jain03 — CHANGES_REQUESTED** [review]

**Comment by VaibhavUpreti** [comment]
Screenshot 2022-12-21 at 12 05 25 AM

**Comment by VaibhavUpreti** [comment]
@tachyons @vedant-jain03 
Can you have a look at this, I will then add the two new notification events in PR #3396 
customising notification events.. with test cases

**Review by vedant-jain03 — APPROVED** [review]
@VaibhavUpreti Awesome work 💯 
cc @tachyons

**Comment by VaibhavUpreti** [comment]
all specs passing @tachyons 👍

File: spec/requests/api/v1/authentication_controller/forgot_password_spec.rb [hunk 4]
```diff
@@ -20,11 +20,12 @@
         # creates a test user
         user = FactoryBot.create(:user)
         post "/api/v1/password/forgot", params: { email: user.email }, as: :json
+        @user = user
       end
 
       it "returns status 200 and should send reset password instructions" do
         expect(response).to have_http_status(:ok)
-        expect(Devise::Mailer).to send_email(:reset_password_instructions)
+        expect(response.parsed_body["message"]).to eq("password reset instructions sent to #{@user.email}")
```
Comment (tachyons) [inline spec/requests/api/v1/authentication_controller/forgot_password_spec.rb:28]: Remove this change as it is not related and false positive

Comment (VaibhavUpreti) [reply spec/requests/api/v1/authentication_controller/forgot_password_spec.rb]: This will have to rebased after service spec PR is merged as I have changed the logic of service... then will refactor the service component(method). Also would have to add one test case.
